articles/112_generated_interfaces_cpp.md

```diff
@@ -104,6 +104,13 @@ All other constants are declared as `static const` members in the struct and the
 ### Constructors
 
 The *default constructor* initializes all members with their default value.
+In some cases this may not be desirable, since these fields will often be immediately overwritten with user-provided values.
+Therefore, the constructor takes an optional directive of type `enum rcl_msg_init_type` to control how initialization is done:
```

Comment:
The data structures must not be coupled with the ROS client library. One goal is that they are usable as plain data classes without requiring the ROS client libraries.

Such an enum needs to be defined in a package which is a runtime dependency of any message package. Since we don't want message packages to runtime depend on the generator package it needs to be in a separate package (similar as `roscpp_core` in ROS 1).